servicetalk-http-api/src/test/java/io/servicetalk/http/api/HeaderUtilsTest.java

```diff
@@ -273,6 +273,17 @@ public void validateToken() {
         }
     }
 
+    @Test
+    public void pathMatches() {
+        assertTrue(HeaderUtils.pathMatches("/a/b/c", "/a/b/c"));
+        assertTrue(HeaderUtils.pathMatches("/a/b/cxxxx", "/a/b/c"));
+        assertTrue(HeaderUtils.pathMatches(new StringBuilder("/a/b/c"), new StringBuilder("/a/b/c")));
```

Comment:
I was surprised that this implements the `CharSequence` interface. It relies on Object.hashCode & Object.equals because its mutable, but then it acts as a String too!